pkg/image/docker/manifest.go

```diff
@@ -1,65 +1,39 @@
 package docker
 
 import (
+	"bytes"
 	"encoding/json"
 	"fmt"
-	"os"
 
-	"github.com/anchore/stereoscope/internal/log"
-	"github.com/anchore/stereoscope/pkg/file"
-	"github.com/google/go-containerregistry/pkg/name"
 	"github.com/google/go-containerregistry/pkg/v1/tarball"
 )
 
-// extractManifest is helper function for extracting and parsing a docker image manifest (V2) from a docker image tar.
-func extractManifest(tarPath string) (tarball.Manifest, error) {
-	f, err := os.Open(tarPath)
-	if err != nil {
-		return nil, err
-	}
-
-	defer func() {
-		err := f.Close()
-		if err != nil {
-			log.Errorf("unable to close tar file (%s): %w", f.Name(), err)
-		}
-	}()
-
-	var manifest tarball.Manifest
-	manifestReader, err := file.ReaderFromTar(f, "manifest.json")
-	if err != nil {
-		return nil, err
-	}
+type manifest struct {
+	raw    []byte
+	parsed tarball.Manifest
+}
 
-	if err := json.NewDecoder(manifestReader).Decode(&manifest); err != nil {
-		return nil, err
+// newManifest creates a new manifest struct from the given Docker archive manifest bytes
+func newManifest(raw []byte) (manifest, error) {
+	var parsed tarball.Manifest
+	if err := json.NewDecoder(bytes.NewReader(raw)).Decode(&parsed); err != nil {
```

Comment:
Just curious — why did you land on `json.NewDecoder(bytes.NewReader(raw)).Decode(&parsed)` instead of `json.Unmarshal(raw, &parsed)`?

Reply:
fingers ahead of thoughts I presume.... (I'll change this)